packages/core/src/amazonq/commons/controllers/contentController.ts

```diff
@@ -3,20 +3,43 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
 import { Position, TextEditor, window } from 'vscode'
 import { getLogger } from '../../../shared/logger'
+import { Commands } from '../../../shared'
 
 export class EditorContentController {
-    public insertTextAtCursorPosition(
+    /* *
+     *  Insert the Amazon Q chat written code to the cursor position
+     *  Add current intentation to the next few lines of the recommendation
+     * @param text the raw text from Amazon Q chat
+     * @param trackCodeEdit callback to track user edits
+     */
+    public async insertTextAtCursorPosition(
         text: string,
         trackCodeEdit: (editor: TextEditor, cursorStart: Position) => void
     ) {
         const editor = window.activeTextEditor
         if (editor) {
             const cursorStart = editor.selection.active
+            const indentRange = new vscode.Range(new vscode.Position(cursorStart.line, 0), cursorStart)
+            // use the user editor intent if the position to the left of cursor is just space or tab
+            // otherwise indent with empty space equal to the intent at this position
+            let indent = editor.document.getText(indentRange)
+            if (indent.trim().length !== 0) {
```

Comment:
also do trimStart() here?

Reply:
If string.trimStart() is non-empty, string.trim() will also be non-empty. Same .

Reply:
The intention is to detect if the indent is purely space or tab, if it is, then we use the same indent. Otherwise we fill with space of same length

Reply:
There are cases when this range has trailing spaces, is it working expected in this logic?

Reply:
https://github.com/user-attachments/assets/e977871a-47e3-4eae-86dd-b980e1307ba3

Yes here is one example when the range has trailing spaces.